src/c++/perf_analyzer/test_concurrency_manager.cc

```diff
@@ -223,6 +220,19 @@ class TestConcurrencyManager : public TestLoadManagerBase,
           doctest::Approx(params_.max_concurrency).epsilon(0.25));
     }
   }
+
+
+  std::shared_ptr<SequenceManager> MakeSequenceManager(
+      const uint64_t start_sequence_id, const uint64_t sequence_id_range,
+      const size_t sequence_length, const bool sequence_length_specified,
+      const double sequence_length_variation, const bool using_json_data,
+      std::shared_ptr<DataLoader> data_loader) override
+  {
+    return std::make_shared<MockSequenceManager>(
```

Comment:
It looks like before this PR, non-mock SequenceManager was used in TestConcurrencyManager--if so, why switch to MockSequenceManager?

Reply:
Seems that is the path we are going. Mock out all of the subclasses with full functionality and only change what is needed. By default our mock classes will behave exactly the same as the parent.

Reply:
But I would say if we have a choice between a `SequenceManager` and a `MockSequenceManager`, might as well stick to `SequenceManager`. I only use a mock when I need some mocking capability (counting calls, checking parameters, disabling/changing logic.

Reply:
I don't have that choice anymore. If we want tests to be able to be deterministic, we need to be able to change the behavior and bypass Infer(), which is what I am doing now.